File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.cpp
class Solution {
public:
// start = 0 and end =0;
// if number is 0 flips --
// once flips become po
    int longestOnes(vector<int>& nums, int k) {
        int start = 0;
        int end = 0;
        int zeroCount = 0;
        int maxSize = 0;

        while(end<nums.size()){
            if(nums[end]==0) zeroCount++;
            while(zeroCount>k){
                if(nums[start]==0) zeroCount--;
                start++;
            }
                
            maxSize = max(maxSize,end-start+1);
            end++;
        }
        return maxSize;
        
    }
};

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
       int count =0;
       int left = 0;
       int right =0;
       int ans = 0;
       int n = nums.length;
       while(right<n){
        if(nums[right]==0) count++;
        while(count>k){
            if(nums[left]==0) count--;
            left++;
        }
         ans = Math.max(ans,right+1-left);
         right++;
       }
       return ans;
    }
}